### PR TITLE
Move test from docker-library/official-images

### PR DESCRIPTION
**Pull request by tianon**
This test isn't useful for "interoperability" like some of the language-specific tests, and is really very specific to this particular image (and we only use it as a tool to roughly validate that we've packaged the software correctly), so it makes more sense for it to live here.

### CODE REVIEW

**Review by acburdine — APPROVED** [review]

**Comment by tianon** [comment]
Might be worth merging https://github.com/docker-library/official-images/pull/14757 first so we can double check this actually works correctly (although I'm not worried).

File: .test/tests/ghost-basics/run.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		-e MYSQL_RANDOM_ROOT_PASSWORD=1 \
+		-e MYSQL_USER="$mysqlUser" \
+		-e MYSQL_PASSWORD="$mysqlPassword" \
+		-e MYSQL_DATABASE="$mysqlDatabase" \
+		"$mysqlImage"
+)"
+trap "docker rm -vf $mysqlCid > /dev/null" EXIT
+
+# "Unknown database error" / "ECONNREFUSED" (and Ghost just crashing hard)
+. "$dir/../../retry.sh" --tries 30 'docker exec -i -e MYSQL_PWD="$mysqlPassword" "$mysqlCid" mysql -h127.0.0.1 -u"$mysqlUser" --silent "$mysqlDatabase" <<<"SELECT 1"'
```
Comment (yosifkit) [inline .test/tests/ghost-basics/run.sh:36]: >```
>/home/runner/work/ghost/ghost/.test/tests/ghost-basics/run.sh: line 36: /home/runner/work/ghost/ghost/.test/tests/ghost-basics/../../retry.sh: No such file or directory
>	'ghost-basics' [4/4]...failed
>```

🤔 what do we do about these "lib" scripts?

Comment (tianon) [reply .test/tests/ghost-basics/run.sh:36]: :sob:

**Comment by tianon** [comment]
Gah, we also need to update the @docker-library-bot automation to look for this and run these tests as well (or finally do the hairy work to try and GitHub-Actions-ize those, which is ... a little complicated).

**Comment by yosifkit** [comment]
Going to close for now. We need to redesign the test harness a bit to make this work (possibly consider moving to a "real" testing framework or something).

**Pull request closed without merging**